xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
CTimer::GetElapsedMilliseconds() is a good example that underlines my statement. It won't work reliable regardless the lock. It implements exactly the scenario I gave above:

```
float CTimer::GetElapsedMilliseconds() const
{
  if (!IsRunning())
    return 0.0f;
 
 !!! no guarantees that we are still running. thread could have terminated already
  return (float)(XbmcThreads::SystemClockMillis() - (m_endTime - m_timeout));
}
```